homeassistant/components/light/__init__.py

```diff
@@ -288,10 +288,20 @@ def state_attributes(self):
                 if value:
                     data[attr] = value
 
-            if ATTR_RGB_COLOR not in data and ATTR_XY_COLOR in data and \
-               ATTR_BRIGHTNESS in data:
-                data[ATTR_RGB_COLOR] = color_util.color_xy_brightness_to_RGB(
-                    data[ATTR_XY_COLOR][0], data[ATTR_XY_COLOR][1],
-                    data[ATTR_BRIGHTNESS])
+            data[ATTR_RGB_COLOR] = self._get_light_rgb_color(data)
```

Comment:
Why did you extract this into a method? It does not seem to be used by anything else. Please revert this.